pandas/tests/extension/base/return_types.py

```diff
@@ -0,0 +1,60 @@
+import pytest
```

Comment:
My previous comment was actually about moving them to a file in `tests/arrays/masked/ ` (so to share the test with the different nullable dtypes). 

Now, I am also fine to move it to the base extension tests as you did here, but then we should add this to *all* EAs. If it's only activated for nullable floating/integer/boolean as is done now, then it should move to `tests/arrays/masked/`, as it's not a general test for all EAs. 

Making it truly generic is probably a bit annoying (you need to know the exact expected type for the scalars), but so maybe the subclasses can override `get_native_dtype` where needed? Or maybe it can be an attribute on the test class (if it's always a single type for each dtype, then that's simpler)

Reply:
> My previous comment was actually about moving them to a file in `tests/arrays/masked/ ` (so to share the test with the different nullable dtypes).
>
> Now, I am also fine to move it to the base extension tests as you did here, but then we should add this to _all_ EAs. If it's only activated for nullable floating/integer/boolean as is done now, then it should move to `tests/arrays/masked/`, as it's not a general test for all EAs.

I did misunderstand but this makes sense

> Making it truly generic is probably a bit annoying (you need to know the exact expected type for the scalars), but so maybe the subclasses can override `get_native_dtype` where needed? Or maybe it can be an attribute on the test class (if it's always a single type for each dtype, then that's simpler)

This sounds like the way to go. I like not having to keep a registry of the mapping between scalars and native types in a single place